File: user/src/main/java/com/kjj/user/entity/UserPolicy.java
package com.kjj.user.entity;

import com.kjj.user.dto.user.DatePolicyDto;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class UserPolicy {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private boolean monday;
    private boolean tuesday;
    private boolean wednesday;
    private boolean thursday;
    private boolean friday;

    private Long defaultMenu;

    public static UserPolicy createNewUserPolicy() {
        return new UserPolicy(
                null,
                false,
                false,
                false,
                false,
                false,
                null
        );
    }

    public void setPolicy(DatePolicyDto dto) {
        monday = dto.isMonday();
        tuesday = dto.isTuesday();
        wednesday = dto.isWednesday();
        thursday = dto.isThursday();
        friday = dto.isFriday();
    }

    public void setDefaultMenu(Long id) {
        defaultMenu = id;
    }
    public void clearDefaultMenu() {
        defaultMenu = null;
    }
}

File: user/src/main/kotlin/com/kjj/user/entity/UserPolicy.kt
package com.kjj.user.entity

import com.kjj.user.dto.user.DatePolicyDto
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class UserPolicy(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,
    var monday: Boolean,
    var tuesday: Boolean,
    var wednesday: Boolean,
    var thursday: Boolean,
    var friday: Boolean,
    var defaultMenu: Long?
) {
    companion object {
        fun createNewUserPolicy(): UserPolicy {
            return UserPolicy(
                null,
                false,
                false,
                false,
                false,
                false,
                null
            )
        }
    }

    fun setPolicy(dto: DatePolicyDto) {
        monday = dto.monday
        tuesday = dto.tuesday
        wednesday = dto.wednesday
        thursday = dto.thursday
        friday = dto.friday
    }
    fun setDefaultMenu(id: Long) {
        defaultMenu = id
    }
}